### PR TITLE
Add group trigger deletion functionality to the module

### PR DESCRIPTION
**Pull request by dkorn**

### CODE REVIEW

**Comment by dkorn** [comment]
@cben @enoodle @kwoodson please review

File: library/hawkular_alerts_group.py [hunk 1]
```diff
@@ -89,13 +89,38 @@
...
+        """
+        try:
+            self.client.get_trigger(group_id)
+            try:
+                self.client.delete_group_trigger(group_id)
+                self.changed = True
+                return dict(
+                    msg="Successfully deleted group trigger {group_id}".format(group_id=group_id),
+                    changed=self.changed)
+            except Exception as e:
```
Comment (enoodle) [inline library/hawkular_alerts_group.py]: Why do you need two try block? you can have multiple except blocks for one try block. Also, why do you raise for `HTTPError` with code different than 404 but for a general error you call `self.module.fail_json` ? This different handling is confusing.

**Review by enoodle — COMMENTED** [review]
I have one comment about the error handling:

File: library/hawkular_alerts_group.py (hunk 1 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py]: > Why do you need two try block?

I'll drop one of the `try`s, thanks.

> Also,  why do you raise for HTTPError with code different than 404 but for a general error you call self.module.fail_json

If the group trigger doesn't exist (`HTTPError` 404) it isn't a fail in the ansible run, actually this is the desired state (even if no change happen).
If the delete action failed we want the module run to fail.
I'll change the general `raise` into `self.module.fail_json`

**Comment by dkorn** [comment]
@cben PTAL

**Comment by dkorn** [comment]
@cben can you review?

**Comment by cben** [comment]
:+1:

`DOCUMENTATION` string for `state` only mentions creation, perhaps should be updated.

For full idempotency, you still miss updating, right? (ah right #6)

**Comment by dkorn** [comment]
> DOCUMENTATION string for state only mentions creation, perhaps should be updated.

Nice catch, fixed.

**Comment by cben** [comment]
OK, one more DOCUMENTATION catch: it says "the state of the user", was probably copy-pasted from another module.

**Comment by dkorn** [comment]
> OK, one more DOCUMENTATION catch: it says "the state of the user", was probably copy-pasted from another module.

Done :)

**Pull request merged**